cmd/legacy_param_converter.go

```diff
@@ -0,0 +1,155 @@
+			"rename-dir-limit":  flags.RenameDirLimit,
+			"temp-dir":          flags.TempDir,
+			"uid":               flags.Uid,
+		},
+		"foreground": flags.Foreground,
+		"gcs-auth": map[string]interface{}{
+			"anonymous-access":     flags.AnonymousAccess,
+			"key-file":             flags.KeyFile,
+			"reuse-token-from-url": flags.ReuseTokenFromUrl,
+			"token-url":            flags.TokenUrl,
+		},
+		"gcs-connection": map[string]interface{}{
+			"billing-project":               flags.BillingProject,
+			"client-protocol":               string(flags.ClientProtocol),
+			"custom-endpoint":               flags.CustomEndpoint,
+			"experimental-enable-json-read": flags.ExperimentalEnableJsonRead,
+			"http-client-timeout":           flags.HttpClientTimeout,
+			"limit-bytes-per-sec":           flags.EgressBandwidthLimitBytesPerSecond,
+			"limit-ops-per-sec":             flags.OpRateLimitHz,
+			"max-conns-per-host":            flags.MaxConnsPerHost,
+			"max-idle-conns-per-host":       flags.MaxIdleConnsPerHost,
+			"sequential-read-size-mb":       flags.SequentialReadSizeMb,
+		},
+		"gcs-retries": map[string]interface{}{
+			"max-retry-sleep": flags.MaxRetrySleep,
+			"multiplier":      flags.RetryMultiplier,
+		},
+		"implicit-dirs": flags.ImplicitDirs,
+		"list": map[string]interface{}{
+			"kernel-list-cache-ttl-secs": flags.KernelListCacheTtlSeconds,
+		},
+		"logging": map[string]interface{}{
+			"file-path": flags.LogFile,
+			"format":    flags.LogFormat,
+		},
+		"metadata-cache": map[string]interface{}{
+			"deprecated-stat-cache-capacity":          flags.StatCacheCapacity,
+			"deprecated-stat-cache-ttl":               flags.StatCacheTTL,
+			"deprecated-type-cache-ttl":               flags.TypeCacheTTL,
+			"enable-nonexistent-type-cache":           flags.EnableNonexistentTypeCache,
+			"experimental-metadata-prefetch-on-mount": flags.ExperimentalMetadataPrefetchOnMount,
+		},
+		"metrics": map[string]interface{}{
+			"stackdriver-export-interval": flags.StackdriverExportInterval,
+		},
+		"monitoring": map[string]interface{}{
+			"experimental-opentelemetry-collector-address": flags.OtelCollectorAddress,
+		},
+		"only-dir": flags.OnlyDir,
+	}
+
+	// Use decoder to convert flagStorage to cfg.Config.
+	decoderConfig := &mapstructure.DecoderConfig{
+		DecodeHook: cfg.DecodeHook(),
+		Result:     resolvedConfig,
+		TagName:    "yaml",
+	}
+	decoder, err := mapstructure.NewDecoder(decoderConfig)
+	if err != nil {
+		return nil, fmt.Errorf("mapstructure.NewDecoder: %v", err)
+	}
+	// Decoding flags.
+	err = decoder.Decode(structuredFlags)
+	if err != nil {
+		return nil, fmt.Errorf("decoder.Decode(structuredFlags): %v", err)
+	}
+
+	// If config file is not present, no need to decode it.
+	if legacyConfig == nil || reflect.DeepEqual(*legacyConfig, config.MountConfig{}) {
+		return resolvedConfig, nil
+	}
+
+	// Save overlapping flags in a map to override the config value later.
+	overlapFlags := map[string]interface{}{
```

Comment:
How about getting rid of the map altogether like this:

if c.IsSet("log-file") {
  resolvedConfig.Logging.FilePath = cfg.ResolvedPath(resolvedConfig.Logging.FilePath)
}

and so on?

Reply:
The flag values in resolvedConfig will be over-witten by the  legacy mount config values so we need to store the resolved values from flags somewhere to over-write.